task.ps1

```diff
@@ -0,0 +1,133 @@
+$location = "uksouth"
+$resourceGroupName = "mate-azure-task-19"
+
+$virtualNetworkName = "todoapp"
+$vnetAddressPrefix = "10.20.30.0/24"
+$webSubnetName = "webservers"
+$webSubnetIpRange = "10.20.30.0/26"
+$mngSubnetName = "management"
+$mngSubnetIpRange = "10.20.30.128/26"
+
+$sshKeyName = "linuxboxsshkey"
+$sshKeyPublicKey = Get-Content "~/.ssh/id_rsa.pub"
+
+$vmImage = "Ubuntu2204"
+$vmSize = "Standard_B1s"
+$webVmName = "webserver"
+$jumpboxVmName = "jumpbox"
+
+$repoUrl = "git@github.com:gaupt/azure_task_19_deploy_web_app.git"
+
+$acrName = "mateacr"  # Make sure the name is globally unique
+$acrSku = "Basic"
+$appName = "mate-webapp"
+$planName = "mate-appservice-plan"
+$registryName = "task-19-todoapp"
+
+Write-Host "Creating a resource group $resourceGroupName ..."
+New-AzResourceGroup -Name $resourceGroupName -Location $location
+
+Write-Host "Creating web network security group..."
+$webHttpRule = New-AzNetworkSecurityRuleConfig -Name "web" -Description "Allow HTTP" `
+   -Access Allow -Protocol Tcp -Direction Inbound -Priority 100 -SourceAddressPrefix `
+   Internet -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 80,443
+$webNsg = New-AzNetworkSecurityGroup -ResourceGroupName $resourceGroupName -Location $location -Name `
+   $webSubnetName -SecurityRules $webHttpRule
+
+Write-Host "Creating mngSubnet network security group..."
+$mngSshRule = New-AzNetworkSecurityRuleConfig -Name "ssh" -Description "Allow SSH" `
+   -Access Allow -Protocol Tcp -Direction Inbound -Priority 100 -SourceAddressPrefix `
+   Internet -SourcePortRange * -DestinationAddressPrefix * -DestinationPortRange 22
+$mngNsg = New-AzNetworkSecurityGroup -ResourceGroupName $resourceGroupName -Location $location -Name `
+   $mngSubnetName -SecurityRules $mngSshRule
```

Comment:
It is not required to create this resources.

Reply:
fix